feat(react-scheduler): support scheduler localization capability (#1985)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/localization.md

# React Scheduler - Localization

The React Scheduler provides an API that allows you to format dates and localize messages.

**Format Dates**

Scheduler UI plugins can display dates that are formatted depending on the locale. To specify the locale, assign one of the [supported formats](https://tools.ietf.org/html/rfc5646) to the [Scheduler](../reference/scheduler.md) component's `locale` property.

**Localize Messages**

Most UI plugins have the `messages` option that accepts localized messages.

## Basic Setup

The following example demonstrates how to localize the Scheduler by specifying the `messages` and `locale` properties.

.embedded-demo({ "path": "scheduler-localization/locale", "showThemeSelector": true })

## Customize Date Formatting

> To customize how dates are formatted in a component, [override it](./fundamentals.md#customize-the-appearance) and set the `formatDate` property to a function that formats dates. The following example illustrates the case. Note that in this example, we use the [Moment.js](https://momentjs.com/) library to work with dates.

Reviewer: ```suggestion
To customize how dates are formatted in a component, [override it](./fundamentals.md#customize-the-appearance) and set the `formatDate` property to a function that formats dates. The following example illustrates these actions. Note that in this example, the [Moment.js](https://momentjs.com/) library is used to work with dates.
```

Reviewer: override it - it what, specify

Reviewer 2: ```suggestion
To customize how dates are formatted in a component, [override the component](./fundamentals.md#customize-the-appearance) and set the `formatDate` property to a function that formats dates. The following example illustrates these actions. Note that in this example, the [Moment.js](https://momentjs.com/) library is used to work with dates.
```

# React Scheduler - Localization

The React Scheduler provides a simple localization API. All the scheduler texts are separated to two parts:
- format dates;
- localization messages.

**Format Dates**

The UI plugins of the Scheduler control often render formated date texts dependent on locale. To specify special locale set the `locale` property of the [Scheduler](../reference/scheduler.md) component with following [format](https://tools.ietf.org/html/rfc5646).

**Localization Messages**

> Almost each UI plugin accepts the `messages` option bag that specifies localized messages. These messages specifies special support texts.

Reviewer 2: ```suggestion
Most UI plugins have the `messages` option that accepts localized messages.
```

## Basic Setup

The following example demonstrates how to change the scheduler localization by specify `messages` and `locale` properties.

.embedded-demo({ "path": "scheduler-localization/locale", "showThemeSelector": true })

## Customize Date Formating

> To change built-in date formating texts you should override our components and pass into them custom date formating function into a `formatDate` property. In the following example we use full week day names in the day scale and use times with seconds in the time scale. In the sample we create a custom formating functions and use a `moment.js` library.

Reviewer 2: ```suggestion
To customize how dates are formatted in a component, [override it](./fundamentals.md#customize-the-appearance) and set the `formatDate` property to a function that formats dates. The following example illustrates the case. Note that in this example, we use the [Moment.js](https://momentjs.com/) library to work with dates.
```

.embedded-demo({ "path": "scheduler-localization/custom-formating", "showThemeSelector": true })

> NOTE: How to override built-in UI components see the [Fundamentals guide](./fundamentals.md#customize-the-appearance).

Reviewer 2: ```suggestion

```

We are giving this link when instructing to override the component.
